src/number_theory/divisors.lean

```diff
@@ -248,4 +248,23 @@ lemma prod_divisors_prime_pow {α : Type*} [comm_monoid α] {k p : ℕ} {f : ℕ
   ∏ x in (p ^ k).divisors, f x = ∏ x in range (k + 1), f (p ^ x) :=
 @sum_divisors_prime_pow (additive α) _ _ _ _ h
 
+lemma finset_div_eq_filter (n : ℕ+) :
```

Comment:
I think I would swap the two sides, make this a simp lemma, and call it `filter_dvd_eq_divisors`.
What do you think?

Reply:
Also, can't you prove this without splitting, by using `finset.ext` and then `mem_divisors`, `mem_filter`, and `mem_range`?
At least, simping with those before splitting seems to deduplicate some of the proof.

Reply:
Oh, the proof is now much better, thank you! I also swapped the two sides and added `@[simp]`.

Reply:
Could you please also update the name?

Reply:
Oh, of course, sorry.